cli/tests/wpt_testharnessconsolereporter.js

```diff
@@ -0,0 +1,119 @@
+const noColor = globalThis.Deno?.noColor ?? true;
+const enabled = !noColor;
+
+function code(open, close) {
+  return {
+    open: `\x1b[${open.join(";")}m`,
+    close: `\x1b[${close}m`,
+    regexp: new RegExp(`\\x1b\\[${close}m`, "g"),
+  };
+}
+
+function run(str, code) {
+  return enabled
+    ? `${code.open}${str.replace(code.regexp, code.open)}${code.close}`
+    : str;
+}
+
+function red(str) {
+  return run(str, code([31], 39));
+}
+
+export function green(str) {
+  return run(str, code([32], 39));
+}
+
+export function yellow(str) {
+  return run(str, code([33], 39));
+}
+
+const testResults = [];
+const testsExpectFail = JSON.parse(Deno.args[0]);
+
+window.add_result_callback((test) => {
+  const expectFail = testsExpectFail.includes(test.name);
+  let simpleMessage = `test ${test.name} ... `;
+  switch (test.status) {
+    case 0:
+      if (expectFail) {
+        simpleMessage += red("ok (expected fail)");
+      } else {
+        simpleMessage += green("ok");
+      }
+      break;
+    case 1:
+      if (expectFail) {
+        simpleMessage += yellow("failed (expected)");
+      } else {
+        simpleMessage += red("failed");
+      }
+      break;
+    case 2:
+      if (expectFail) {
+        simpleMessage += yellow("failed (expected)");
+      } else {
+        simpleMessage += red("failed (timeout)");
+      }
+      break;
+    case 3:
+      if (expectFail) {
+        simpleMessage += yellow("failed (expected)");
+      } else {
+        simpleMessage += red("failed (incomplete)");
+      }
+      break;
+  }
+
+  console.log(simpleMessage);
+
+  testResults.push({
+    name: test.name,
+    passed: test.status === 0,
+    expectFail,
+    message: test.message,
+    stack: test.stack,
+  });
+});
```

Comment:
If you do the above...

```suggestion
  testResults.push({
    name,
    passed: status === 0,
    expectFail,
    message,
    stack,
  });
});
```